src/AutoMapper/IMemberConfigurationExpression.cs

```diff
@@ -199,6 +199,13 @@ void ResolveUsing<TValueResolver>()
         /// The destination member being configured.
         /// </summary>
         MemberInfo DestinationMember { get; }
+
+        /// <summary>
+        /// Apply a transformation function after any resolved destination member value with the given type
+        /// </summary>
+        /// <typeparam name="TValue">Value type to match and transform</typeparam>
+        /// <param name="transformer">Transformation expression</param>
+        void AddTransform<TValue>(Expression<Func<TValue, TValue>> transformer);
```

Comment:
Maybe a non generic overload that just uses TMember.

Reply:
I don't know if we need an overload as much as just get rid of this version. I don't remember why I put a transform where you could specify a different type. Transformers get applied after resolving/mapping so it's always going to be of the destination member type.

Reply:
After resolving and before mapping. And I think you meant for TValue to be the type of the resolved value. Which doesn't have to be TMember. Right? :)

Reply:
Ah then I wonder if I have value transformations in the right place. I think it's really before you assign it, you transform it but I have it as part of resolving.

Reply:
I moved the value transformations to after mapping and right before setting.

Reply:
I like it. I've always thought AfterPropertyMap is a good idea :)

Reply:
Ha well maybe this will take care of people's scenarios.